util/collections/machine_collection.go

```diff
@@ -176,3 +176,25 @@ func (s Machines) Names() []string {
 	}
 	return names
 }
+
+// SortedByVersion returns the machines sorted by version.
+func (s Machines) SortedByVersion() []*clusterv1.Machine {
+	res := make(util.MachinesByVersion, 0, len(s))
+	for _, value := range s {
+		res = append(res, value)
+	}
+	sort.Sort(res)
+	return res
+}
+
+// LowestVersion returns the lowest version among all the machine with
+// defined versions. If no machine has a defined version it returns an
+// empty string.
+func (s Machines) LowestVersion() string {
```

Comment:
```suggestion
func (s Machines) LowestVersion() *string {
```
So we can just return nil in case it's not found?

Reply:
Yes, it should be possible. I will refactor the function to return a pointer instead.